syntax = "proto3";

package v1;

option go_package = "github.com/thoseJanes/tinyblog/pkg/proto/aiservice/v1";

service AIService{
    rpc generateTitleAndTag (PromptContentRequest) returns (GenerateTitleAndTagResponse) {}
    rpc polishContent (PromptContentRequest) returns (stream PolishContentResponse) {}
    rpc searchPosts (PromptRequest) returns (SearchPostsResponse) {}
    rpc summaryContent (ContentRequest) returns (stream SummaryContentResponse) {}
}

message PromptRequest{
    string prompt = 1;
}

// message SummaryPostsRequest{
//     string prompt = 1;
//     repeated string ids = 2;
// }



message SearchPostsResponse{
    repeated string ids = 1;
    string evaluation = 2;
}

message ContentRequest{
    string content = 1;
}

message PromptContentRequest{
    string prompt = 1;
    string content = 2;
}

message GenerateTitleAndTagResponse{
    string title = 1;
    repeated string tags = 2;
}



message PolishContentResponse{
    string contentChunk = 1;
}

message SummaryContentResponse{
    string contentChunk = 1;
}